src/GraphRequest.ts

```diff
@@ -501,19 +494,26 @@ export class GraphRequest {
 	 * @returns The same GraphRequest instance that is being called with
 	 */
 	public query(queryDictionaryOrString: string | KeyValuePairObjectStringNumber): GraphRequest {
-		const otherURLQueryParams = this.urlComponents.otherURLQueryParams;
+		let paramKey: string;
+		let paramValue: string | number;
 		if (typeof queryDictionaryOrString === "string") {
-			const querySplit = queryDictionaryOrString.split("=");
-			const queryKey = querySplit[0];
-			const queryValue = querySplit[1];
-			otherURLQueryParams[queryKey] = queryValue;
+			const indexOfFirstEquals = queryDictionaryOrString.indexOf("="); // The query key-value pair must be split on the first equals sign to avoid error while parsing nested query parameters
+			const qParams = [queryDictionaryOrString.substring(0, indexOfFirstEquals), queryDictionaryOrString.substring(indexOfFirstEquals + 1, queryDictionaryOrString.length)];
+			paramKey = qParams[0];
+			paramValue = qParams[1];
 		} else {
 			for (const key in queryDictionaryOrString) {
 				if (queryDictionaryOrString.hasOwnProperty(key)) {
-					otherURLQueryParams[key] = queryDictionaryOrString[key];
+					paramKey = key;
+					paramValue = queryDictionaryOrString[key];
 				}
 			}
 		}
+		if (oDataQueryNames.indexOf(paramKey) !== -1) {
+			this.urlComponents.oDataQueryParams[paramKey] = paramKey === "$expand" || paramKey === "$select" || paramKey === "$orderby" ? (this.urlComponents.oDataQueryParams[paramKey] ? this.urlComponents.oDataQueryParams[paramKey] + "," + paramValue : paramValue) : paramValue;
```

Comment:
The existing code contains a [addCsvQueryParameter function](https://github.com/microsoftgraph/msgraph-sdk-javascript/blob/870477b4eefd5ba7c5838cbd64280c5cc2cbc362/src/GraphRequest.ts#L193)  which is called from the .select, expand and orderby query functions. The function does a comma separated concatenation for multiple values of a given key. The idea to introduce this in the query and thereby the parsePath function is to maintain consistency. For example, if we use client.api("/me?$select=id").select("name") then the query parameter part of the output would be  "/me?$select= id,name". But client.api("/me?$select=id&$select=name") would output the query parameter part as "/me?$select=name" that is replace the first query param value. Therefore, I added the comma separated concatenation from the addCsvQueryParameter to query function so that different functions to build a query string behave the same way.  

We do support  parsing "http://host/service/Products?$select=Name,Description&$expand=Category($select=Name)". The query parameter is separated on the first equals and nested part is maintained in the value part as is. I ran test case and got the expected output ->  

>"/Products$select=Name,Description&$expand=Category($select=Name)":"https://graph.microsoft.com/v1.0/Products?$select=Name,Description&$expand=Category($select=Name)" .